Add howto rename master to main branches (#3019)
alphagov/govuk-developer-docs: source/manual/howto-switch-default-branch.html.md

---
owner_slack: "#govuk-developers"

> title: How to switch a git default branch from master to main

Reviewer: ```suggestion
title: How to change the default branch of a Git repo
```

It might be nice to keep the docs generic enough that they can stay around - but maybe that's unnecessary...?

Reviewer 2: We'd need to update release to accept any input for default branch if we choose to do that. It's a pretty trivial UI change though. I decided against having it being free text as it seemed very unlikely to need other options.

Reviewer: That's true, it's probably not worth it really.

Reviewer: I'd still suggest capitalisation of `Git` and use the phrase `change` rather than `switch` though (but this is personal preference).

Author: Ok left it specific, but upper case G on Git.

section: GitHub
layout: manual_layout
parent: "/manual.html"
---

These steps are for switching the default branch of a repo to `main`.

1. Rename the `master` branch in GitHub to `main` for that repo. See [instructions from Github for renaming an existing default branch](https://github.com/github/renaming#renaming-existing-branches).
1. Update the [Release app](https://release.publishing.service.gov.uk/applications/):
  - Go to the app entry from the [listing](https://release.publishing.service.gov.uk/applications/).
  - Hit the `Edit` tab.
  - Switch under the dropdown for `GitHub repository default branch`.

> 1. [Reset the Jenkins build number](https://docs.publishing.service.gov.uk/manual/test-and-build-a-project-on-jenkins-ci.html#fixing-the-build-number).

Reviewer: We might want to be explicit here on where to get the build number from, i.e. it has to be the latest `master` build number plus 1.

Author: Great suggestion :+1:

Reviewer: I think you may have added `(the last `master` build +1)` to the wrong line, I think it should be this one rather than the one above.

Author: 🤦

> 1. Delete the master branch locally, otherwise you might rebase the wrong branch by mistake!

Reviewer: GitHub includes some instructions on how to do this that it might be useful to link to: https://docs.github.com/en/github/administering-a-repository/renaming-a-branch#updating-a-local-clone-after-a-branch-name-changes

Rather than deleting the `master` branch locally, it renames it to `main` and keeps it up to date with origin.

Author: Ooooh! 😍

## Additional things to check

> - Update references in Jenkins to pulling down the `master` branch of the app.

Reviewer: It might be worth adding some instructions on how to use the GitHub search feature to find these. (https://github.com/alphagov/govuk-puppet/search?l=HTML%2BERB&q=master)

Author: Nice! Done
